### ISSUE TITLE
The zoom in/out function stops working once AutoZoom is selected with large graph data

### ISSUE THREAD

**Issue by rahul-winner** [swimlane/ngx-graph#143]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here
```

**Current behavior**

If we auto-zoom on a graph with large data then zoom stops working. One is not able to zoom in/out once auto-zoom is selected. 

**Expected behavior**

Zoom in/out should work fine even after selecting auto-zoom with large graph.

**Reproduction of the problem**

I could reproduce this problem on demo page as well : https://swimlane.github.io/ngx-graph/
Perform following steps
- select real data
- wait for two minutes (approx) to populate some good amount of data so that screen fills with nodes.
- select auto-zoom (the graph will fit to the screen) or select 'FIT IN VIEW'
- try to zoom in/out

Possibly it happens when zoom level goes below 0.1 Because we could not zoom in below level of 0.1 when using manual zoom through mousewheel.

Kindly let me know if you want elaboration.

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**

NPM = 6.x

* **ngx-graph version:** x.x.x

ngx-graph version = 5.2.1

* **Angular version:** x.x.x

Angular = 7.x (I could reproduce it with 6.x)

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

all

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
all

**Comment by assafsun** [comment]
Can you please elaborate why do you think zoom level got below 0.1?

Thanks.

**Comment by rahul-winner** [comment]
@assafsun on Demo page, when I recreated this issue, I noticed the zoom-level went below 0.1. Please take a look on this image when zoom in/out stops working.

![image](https://user-images.githubusercontent.com/4982751/50626195-dfeb8580-0f52-11e9-8827-87bc9e8f29e0.png)

**Comment by rahul-winner** [comment]
@assafsun did you find some solution to this?

**Comment by rahul-winner** [comment]
Thanks @assafsun for your response.

As I stated in the steps for reproducing this issue. You will need to supply huge data, on demo page if you select 'real data' and wait for 2-3 minutes, it gets filled automatically with bigger graph. Then select "FIT IN VIEW". Check the zoom level if it does not get below 0.1 then wait for some more time to produce more nodes and links. And then again select "FIT IN VIEW".

> In general, you can control and maintains that the zoom will not go between below in your personal app.
We could do this. However, the problem here is within the function which is fitting the graph in the view and their I believe no check on zoom level. In order to fit the bigger graph in the view, it must be reducing level and that is going below 0.1 and consequently causing this problem.

Kindly let me know if you need elaboration further.

**Comment by assafsun** [comment]
After more than 2-3 minutes, I succeeded to reproduce it.
We will keep this issue open.

The required change IMO is to add validation of zoom level between 0.1 - 4 in the graph component.

Thanks @rahul-winner

**Comment by assafsun** [comment]
I added the fix to my latest PR #146

**Comment by rahul-winner** [comment]
thanks @assafsun 

Is it possible to commit it and expedite the process for getting it to released version? We want to freeze the versions for npm packages this week. It will be a great help for us.

Regards, Rahul

**Comment by assafsun** [comment]
That's our goal.

**Comment by marjan-georgiev** [comment]
Merged the PR and released it in version 5.3.0. Please let us know if you're still seeing the issue.

**Comment by rahul-winner** [comment]
Thanks a lot @assafsun @marjan-georgiev for quickly fixing it. It is really helpful. I'll update the ngx-graph to 5.3.0 in our app, test it. Hopefully it should address the problem. 

Thanks again.
